pdata/pprofile/go.mod

```diff
@@ -2,8 +2,6 @@ module go.opentelemetry.io/collector/pdata/pprofile
 
 go 1.21.0
 
-toolchain go1.21.11
```

Comment:
Shouldn't this be `toolchain go1.21.12` rather than remove?

Reply:
we don't have toolchain in any modules other than the ones generated by the builder